tests/geomodel/core/test-geomodel-build.cpp

```diff
@@ -0,0 +1,641 @@
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_interfaces != nb_interface_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of interfaces are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_layers != nb_layer_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of layers are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_surfaces != nb_surface_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of surfaces are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_regions != nb_region_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of regions are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_lines != nb_line_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of lines are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+
+    if( nb_corners != nb_corner_entities )
+    {
+        throw RINGMeshException( "RINGMesh Test",
+            "The numbers of corners are not equal " + geomodel.name()
+                + ": the loaded model is not valid." );
+    }
+    return true;
+}
+
+gmge_id create_geology_entity(
+    const GeologicalEntityType& entity_type, GeoModelBuilder3D& builder )
+{
+    return builder.geology.create_geological_entity( entity_type );
+}
+
+gmme_id create_topology_entity(
+    const MeshEntityType& entity_type, GeoModelBuilder3D& builder )
+{
+    return builder.topology.create_mesh_entity( entity_type );
+}
+
+void build_geomodel( GeoModel3D& geomodel )
+{
+    GeoModelBuilder3D builder( geomodel );
+
+    // We first create the GeoModelGeoglogicalEntity
+    // Create the contacts
+    for( index_t contact = 0; contact < nb_contacts; contact++ )
+    {
+        create_geology_entity( Contact3D::type_name_static(), builder );
+    }
+    /*       builder.geology.create_geological_entity(
+               Contact3D::type_name_static() );
+               */ // the static method type_name_static() is available for each
+    // GeoModelEntity. It returns an EntityType which is a string
+    // corresponding to the Type of the entity.
+
+    // Create the Interfaces
+    for( index_t interface_itr = 0; interface_itr < nb_interfaces;
+         interface_itr++ )
+    {
+        create_geology_entity( Interface3D::type_name_static(), builder );
+        // builder.geology.create_geological_entity(
+        //    Interface3D::type_name_static() );
+    }
+
+    // Create the Layers
+    for( index_t layer = 0; layer < nb_layers; layer++ )
+    {
+        create_geology_entity( Layer3D::type_name_static(), builder );
```

Comment:
In each for block add a ringmesh_unused(layer); to avoid warning in your code. We want 0-warning code on master branch